soc/nordic/soc.yml

```diff
@@ -1,39 +1,73 @@
 family:
-- name: nordic_nrf
-  series:
-  - name: nrf51
-    socs:
-    - name: nrf51822
-  - name: nrf52
-    socs:
-    - name: nrf52805
-    - name: nrf52810
-    - name: nrf52811
-    - name: nrf52820
-    - name: nrf52832
-    - name: nrf52833
-    - name: nrf52840
-  - name: nrf53
-    socs:
-    - name: nrf5340
-      cpuclusters:
-      - name: cpuapp
-      - name: cpunet
-  - name: nrf54l
-    socs:
-    - name: nrf54l15
-      cpuclusters:
-      - name: cpuapp
-  - name: nrf54h
-    socs:
-    - name: nrf54h20
-      cpuclusters:
-      - name: cpuapp
-      - name: cpurad
-      - name: cpuppr
-  - name: nrf91
-    socs:
-    - name: nrf9131
-    - name: nrf9151
-    - name: nrf9160
-    - name: nrf9161
+  - name: nordic_nrf
+    series:
+      - name: nrf51
+        socs:
+          - name: nrf51822
+      - name: nrf52
+        socs:
+          - name: nrf52805
+          - name: nrf52810
+          - name: nrf52811
+          - name: nrf52820
+          - name: nrf52832
+          - name: nrf52833
+          - name: nrf52840
+      - name: nrf53
+        socs:
+          - name: nrf5340
+            cpuclusters:
+              - name: cpuapp
+              - name: cpunet
+      - name: nrf54l
+        socs:
+          - name: nrf54l15
+            cpuclusters:
+              - name: cpuapp
+      - name: nrf54h
+        socs:
+          - name: nrf54h20
+            cpuclusters:
+              - name: cpuapp
+              - name: cpurad
+              - name: cpuppr
+      - name: nrf91
+        socs:
+          - name: nrf9131
+          - name: nrf9151
+          - name: nrf9160
+          - name: nrf9161
+
+# Recovery/erase is only needed once per core.
+run_once:
+  - '--recover':
+      - boards:
+          - /(.+)/nrf5340/cpunet/
+      - boards:
+          - /(.+)/nrf5340/cpuapp/
+          - /(.+)/nrf5340/cpuapp/ns/
+      - boards:
+          - /(.+)/nrf9160/
+          - /(.+)/nrf9160/ns/
+  - '--erase':
+      - boards:
+          - /(.+)/nrf5340/cpunet/
+      - boards:
+          - /(.+)/nrf5340/cpuapp/
+          - /(.+)/nrf5340/cpuapp/ns/
+      - boards:
+          - /(.+)/nrf9160/
+          - /(.+)/nrf9160/ns/
+
+# Prevent resetting the cores whilst flashing multiple images until all images
+# for each core have been flashed, this allows security bits to be set during
+# programming without them interfering with additional flashing processes.
+deferred_reset:
+  - boards:
+      - /(.+)/nrf5340/cpuapp/
+      - /(.+)/nrf5340/cpuapp/ns/
+  - boards:
+      - /(.+)/nrf5340/cpunet/
+  - boards:
+      - /(.+)/nrf9160/
+      - /(.+)/nrf9160/ns/
```

Comment:
why N `boards`? can't they be merged into a single boards list?

Reply:
No, they are each groups. If you imagine a board with an nrf5340 and nrf9160 on it then as a single group it would mean one of those would be reset and the other would not

Reply:
I think we need to make this more intuitive

Reply:
I don't see how that can be done, there needs to be a grouping, you can't know this implicitly for all socs as it's going to vary between different manufacturers

Reply:
then give it a dedicated group field / name.
Right now, it's not intuitive at all that two `boards:` lists actually means two groups.

And explain what a group means.

I have very similar comment which I wrote before seeing this comment.